content/post/52-go-ipfs-0.4.18.md

```diff
@@ -0,0 +1,283 @@
+final, most things you can do via the CLI or HTTP interfaces, can now be done
+through the new API.
+
+Currently there is only one implementation, backed by go-ipfs node, and there are
+plans to start http-api backed one soon. We are also looking into creating RPC
+interface using this API, which could help performance in some use cases.
+
+You can track progress in https://github.com/ipfs/go-ipfs/issues/4498
+
+### CIDv1/Base32 Migration
+
+Currently, IPFS is usually used in browsers by browsing to `https://SOME_GATEWAY/ipfs/CID/...`. There are two significant drawbacks to this approach:
+
+1. From a browser security standpoint, all IPFS "sites" will live under the same
+   origin (SOME_GATEWAY).
+2. From a UX standpoint, this doesn't feel very "native" (even if the gateway is
+   a local IPFS node).
+
+To fix the security issue, we intend to switch IPFS gateway links
+`https://ipfs.io/ipfs/CID` to to `https://CID.ipfs.dweb.link`. This way, the CID
+will be a part of the
+["origin"](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin) so
+each IPFS website will get a separate security origin.
+
+To fix the UX issue, we've been working on adding support for `ipfs://CID/...`
+to web browsers through our
+[ipfs-companion](https://github.com/ipfs/ipfs-companion/) add-on and some new,
+experimental extension APIs from Mozilla. This has the same effect of putting
+the CID in the URL origin but has the added benefit of looking "native".
+
+Unfortunately, origins must be *case insensitive*. Currently, most CIDs users
+see are *CIDv0* CIDs (those starting with `Qm`) which are *always* base58
+encoded and are therefore case-sensitive.
+
+Fortunately, CIDv1 (the latest CID format) supports arbitrary bases using the
+[multibase](https://github.com/multiformats/multibase/) standard. Unfortunately,
+IPFS has always treated equivalent CIDv0 and CIDv1 CIDs as distinct. This means
+that files added with CIDv0 CIDs (the default) can't be looked up using the
+equivalent CIDv1.
+
+This release makes some significant progress towards solving this issue by
+introducing two features:
+
+(1) The previous mentioned `ipfs cid base32` command for converting CID to a
+case intensive encoding required by domain names. This command converts a CID to
+version 1 and encodes it using base32.
+
+(2) A hack to allow locally looking up blocks associated with a CIDv0 CID using
+the equivalent CIDv1 CID (or the reverse). This hack will eventually
+be replaced with a multihash indexed blockstore, which is agnostic to both the
+CID version and multicodec content type.
+
+## Full Changelog
+
+As always, you can find the full (massive) changelog over at ipfs/go-ipfs's Github
+repository:
+https://github.com/ipfs/go-ipfs/blob/master/CHANGELOG.md#go-ipfs-changelog-1
+
+## Contributing
+
+Would you like to help contribute to the go-ipfs project? Join us on Github at https://github.com/ipfs/go-ipfs where you can find out more about the project and how you can help. Thanks!
```

Comment:
Other contributing ideas:
- Check the issues with the help wanted label at the Ready column in our waffle board - https://waffle.io/ipfs/go-ipfs?label=help%20wanted
- Join an IPFS All Hands, introduce yourself and let us know where you would like to contribute or any cool demos of what you've built - https://github.com/ipfs/pm/#all-hands-call
- Join the discussion at http://discuss.ipfs.io/ and help users finding their answers.
- Join the Go Core Dev Team Weekly Sync 🙌🏽 and be part of the Sprint action!